Simplenote/src/main/java/com/automattic/simplenote/models/Tag.java

```diff
@@ -78,55 +76,38 @@ public void setIndex(Integer tagIndex) {
     }
 
     public void renameTo(String tagOld, String tagNew, int index, Bucket<Note> notesBucket) throws BucketObjectNameInvalid {
-        // When old tag ID is equal to new tag hash, tag is being renamed to lexical variation.
-        boolean isOldIdEqualToNewHash = getSimperiumKey().equals(TagUtils.hashTag(tagNew));
+        if (tagOld.equals(tagNew)) return;
+
+        renameTagInNotes(tagOld, tagNew, notesBucket);
```

Comment:
how about we let this whole process exist inside of `TagUtils` - I like how you moved part of it there already, and I'd rather not coupled the notes bucket into the tag bucket if possible, though I acknowledge the inherent challenge presented with the data model.

```java
TagUtils.renameTag(String oldTagName, String newTagName);
```

if we do that then we can let `TagUtils.renameTag()` call `Tag.rename()` and also iterate over all the notes. this `Tag.renameTo` could be a bit simpler. hmm. thinking…